src/Sarif.Converters.UnitTests/SemmleQLConverterTests.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Microsoft.CodeAnalysis.Sarif.Converters;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    [TestClass]
+    public class SemmleQLConverterTests : ConverterTestsBase<SemmleQLConverter>
+    {
+        [TestMethod]
+        public void SemmleQLConverter_SimpleCsv()
+        {
+            var semmleCsvInput = SemmleCsvRecord.BuildDefaultRecord().ToCsv();
+
+            string expected = @"{
+  ""$schema"": ""http://json.schemastore.org/sarif-1.0.0"",
+  ""version"": ""1.0.0"",
+  ""runs"": [
+    {
+      ""tool"": {
+        ""name"": ""Semmle QL""
+      },
+      ""results"": [
+        {
+          ""level"": ""error"",
+          ""message"": ""Message"",
+          ""locations"": [
+            {
+              ""resultFile"": {
+                ""uri"": ""RelativePath"",
+                ""uriBaseId"": ""$srcroot"",
+                ""region"": {
+                  ""startLine"": 1,
+                  ""startColumn"": 2,
+                  ""endLine"": 3,
+                  ""endColumn"": 4
+                }
+              }
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}";
+            RunTestCase(semmleCsvInput, expected, prettyPrint: true);
+        }
+
+        private class SemmleCsvRecord
+        {
+            public string QueryName;
+            public string Description;
+            public string Severity;
+            public string Message;
+            public string RelativePath;
+            public string Path;
+            public int StartLine;
+            public int StartColumn;
+            public int EndLine;
+            public int EndColumn;
+
+            public static SemmleCsvRecord BuildDefaultRecord()
+            {
+                return new SemmleCsvRecord()
+                {
+                    QueryName = QueryNameDefault,
+                    Description = DescriptionDefault,
+                    Severity = SeverityDefault,
+                    Message = MessageDefault,
+                    RelativePath = RelativePathDefault,
+                    Path = PathDefault,
+                    StartLine = StartLineDefault,
+                    StartColumn = StartColumnDefault,
+                    EndLine = EndLineDefault,
+                    EndColumn = EndColumnDefault
+                };
+            }
+
+            public string ToCsv()
+            {
+                return
```

Comment:
Avoids repeated string additions with repeated separator literals.

---
In reply to: [134354565](https://github.com/Microsoft/sarif-sdk/pull/590#discussion_r134354565) [](ancestors = 134354565)